django/recaptcha.md

```diff
@@ -0,0 +1,72 @@
+    - Create an `.env.example` file in your root directory. This will give you a file that can be commited to version control and shared with other developers, so they know what environment variables to set for their own local development. [Here is an example from one of our projects](https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/.env.example#L9), with the keys for Google's test reCAPTCHA.
+    - Once you have your `.env.example` file, copy it to a local `.env` file so that your application can use the environment variables.
+
+3. Configure docker-compose to use the environment file
+    - Reference the `.env` file in the project's root `docker-compose.yml` file. This will enable the app to run locally and use your local environment variables. [Here is an example](https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/docker-compose.yml#L26).
+    - Add the environment variables to `~/tests/docker-compose.yml`, [like so](https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/tests/docker-compose.yml#L13).
+
+Now you can use the public key and private key in your code.
+
+### Implement your code
+There are a few ways of doing this, so [refer to Google](https://developers.google.com/recaptcha/intro) when considering how to do this in your project. For this guide, we are implementing V2 of reCAPTCHA.
+
+1. Configure the view with the public key. [Pass the public key into the template via the context](https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/parserator_web/views.py#L519).
+
+2. Add Google's magic JavaScript and public key to your template.
+    -  We have [an example of how this looks](https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/parserator_web/templates/parserator_web/sign_up.html#L53) in an application. [See Google's documentation](https://developers.google.com/recaptcha/docs/display#auto_render) for other ways to do this, too.
+
+3. Write a helper function to validate the reCAPTCHA with Google's API. Your backend needs to accept the reCAPTCHA response from the frontend, and then send a request to Google to validate the reCAPTCHA response. 
+    - Here's an example of how we've done this with V2: https://github.com/datamade/parserator.datamade.us/blob/bda3201c3d7873916ed4075a2102b5805fad9a3a/parserator_web/views.py#L524.
+    - Here's another example of this with V3: https://github.com/datamade/django-salsa-auth/commit/c8512d030b90762c7d703bfd1630f79d11e10a5e#diff-bfb393f3e832ecb2f6fb86ad35eefa88a87e0d773b3e4a80ce224d83997df815R137
+
+5. Test it manually and automatically. If everything was setup correctly, then you should be able submit the form. 
+
+By this point, everything should be working in your app so that you can change out the test keys and use a real reCAPTCHA.
```

Comment:
I think we can edit this down to reference the Django plug-in as our default implementation, then reference our homespun examples for instances where use of the plug-in is not feasible.

Reply:
Sounds good.

Reply:
I took it all out and linked to our different implementations. Way cleaner and less confusing.